Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -356,33 +357,27 @@ public protocol ChatChannelListControllerDelegate: DataControllerStateDelegate {
     )
     
     /// The controller asks the delegate if the newly inserted `ChatChannel` should be linked to this Controller's query.
-    /// Defaults to `false`
+    /// Defaults to `true`
     /// - Parameters:
     ///   - controller: The controller,
-    ///   - shouldLinkNewChannel: The newly inserted `ChatChannel` instance. This instance is not linked to the controller's query.
+    ///   - shouldAddNewChannelToList: The newly inserted `ChatChannel` instance. This instance is not linked to the controller's query.
+    /// - Returns:
+    ///     `true` if channel should be added to the list of observed channels, `false` if channel doesn't exists in this list.
     func controller(
         _ controller: ChatChannelListController,
-        shouldLinkNewChannel channel: ChatChannel
+        shouldAddNewChannelToList channel: ChatChannel
     ) -> Bool
     
     /// The controller asks the delegate if the newly updated `ChatChannel` should be linked to this Controller's query.
-    /// Defaults to `false`
+    /// Defaults to `true`
     /// - Parameters:
     ///   - controller: The controller,
-    ///   - shouldLinkUpdatedChannel: The newly updated `ChatChannel` instance. This instance is not linked to the controller's query.
+    ///   - shouldListUpdatedChannel: The newly updated `ChatChannel` instance.
+    /// - Returns:
+    ///     `true` if channel should be added to the list of observed channels, `false` if channel doesn't exists in this list.
     func controller(
         _ controller: ChatChannelListController,
-        shouldLinkUpdatedChannel channel: ChatChannel
-    ) -> Bool
-    
-    /// The controller asks the delegate if the newly updated `ChatChannel` should be unlinked from this Controller's query.
-    /// Defaults to `false`
-    /// - Parameters:
-    ///   - controller: The controller,
-    ///   - shouldUnlinkUpdatedChannel: The newly updated `ChatChannel` instance. This instance is linked to the controller's query.
-    func controller(
-        _ controller: ChatChannelListController,
-        shouldUnlinkUpdatedChannel channel: ChatChannel
+        shouldListUpdatedChannel channel: ChatChannel
```

Comment:
If these are breaking changes, shouldn't we deprecate them first? We are already in production 4.0.

Reply:
This wasn't released, only on main

Reply:
Hum okay 🤔 Then if we fix the Channel List tests, it should be LGTM